### ISSUE TITLE
Just a question

### ISSUE THREAD

**Issue by somoza** [tzumby/ex_keccak#31]
Sorry for the use issues for this... Please close it as soon you answer me. 

I'm just curious...  Why did you use rust for this and not native erlang? Did you make some benchmarks or something to make this decision?

**Comment by tzumby** [comment]
Hey @somoza, this is def the right place to ask this kind of questions :). 

At the time I wrote this, there was only a C based Nif that wasn't maintained anymore. From what I could research there was no pure Erlang implementation. Maybe that's changed now ?

In terms of benchmark, I don't think we need to compare Erlang with Rust on those operations. Rust would probably be orders of magnitude faster.

**Comment by somoza** [comment]
Thanks for tour answer! I found that [BCrypt.base]( https://hexdocs.pm/bcrypt_elixir/Bcrypt.Base.html) module uses the Erlang [crypto.erl](https://github.com/erlang/otp/blob/master/lib/crypto/src/crypto.erl)

May this do the work?

**Comment by somoza** [comment]
If so, I would be glad to help on building a prototype using the Erlang library, and besides, compare both with some benchmarking.

**Comment by tzumby** [comment]
Yes, the `bcrypt_elixir` lib does use the native crypto module. Erlang crypto does include sha3, but that won't help with this as the final padding is different and results in a different hash. I supposed one could take the native implementation and change the padding.

**Comment by somoza** [comment]
Yup, you're right. I'm not an Erlang programmer but I will try to add this to the Erlang OTP anyway. Thanks!

**Comment by somoza** [comment]
It depends on the OpenSSL release (Erlang uses its apis), so we have to wait. The other option is to program it on pure erlang as someone did on Rust ^^

https://github.com/openssl/openssl/issues/19304#issuecomment-1263230431
